src/operator/tensor/elemwise_binary_scalar_op_basic.cc

```diff
@@ -53,16 +53,16 @@ static bool BinaryScalarStorageTypeWithDenseResultStorageType(const NodeAttrs& a
                                                               std::vector<int>* in_attrs,
                                                               std::vector<int>* out_attrs)  {
   bool dispatched = false;
+  const double alpha = nnvm::get<double>(attrs.parsed);
   if (common::ContainsOnlyStorage(*in_attrs, kDefaultStorage)) {
     dispatched = storage_type_assign(&out_attrs[0],
-                                     kDefaultStorage,
-                                     dispatch_mode,
-                                     DispatchMode::kFCompute);
-  } else if (dev_mask == kCPU) {
+      kDefaultStorage, dispatch_mode, DispatchMode::kFCompute);
+  } else {
     dispatched = storage_type_assign(&out_attrs[0],
-                                     kDefaultStorage,
-                                     dispatch_mode,
-                                     DispatchMode::kFComputeEx);
+      alpha == 0 ? static_cast<NDArrayStorageType>(in_attrs->at(0)) : kDefaultStorage,
+      dispatch_mode, dev_mask == kCPU ? DispatchMode::kFComputeEx
+                                      : DispatchMode::kFComputeFallback);
+    if (dispatched && dev_mask != kCPU) LogStorageFallback(attrs, dev_mask, in_attrs, out_attrs);
```

Comment:
Whether to log storage fallback should only depend on the assigned dispatch_mode: https://github.com/apache/incubator-mxnet/blob/d2a856a3a2abb4e72edc301b8b821f0b75f30722/src/operator/elemwise_op_common.h#L79